tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,173 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	runningTime := 60 * time.Second // time for which operation should be performed
+	noOfPrefixes := 20              // total number of prefixes which will be watched upon
+	watchPerPrefix := 10            // number of watchers per prefix
+	reqRate := 20                   // put request per second
+	keyPrePrefix := 1000            // max number of keyPrePrefixs for put operation
+
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	prePrefixKeys := generateRandomKeys(10, keyPrePrefix)
+	// map for holding prefixes and values
+	prefixKeysMap := make(map[string][]string)
+	// actual keys for performing put operation
+	keys := make([]string, 0)
+
+	// initialize the map with all prefixes
+	for _, prefix := range prefixes {
```

Comment:
@xiang90 but as i mentioned if we want it to make it robust and more practical we will need this. if somehow we make all prefixes unique, then it becomes as good as watching on a key, and its not something we will want when we want to ensure the watch functionalities under all circumstances.